pentaho-webjars-deployer/src/main/java/org/pentaho/osgi/platform/webjars/WebjarsURLConnection.java

```diff
@@ -151,48 +187,200 @@ private void transform( URL url, PipedOutputStream pipedOutputStream ) throws IO
+
+          moduleName = ( String ) xPath.evaluate("/project/artifactId", pom.getDocumentElement(), XPathConstants.STRING);
+          moduleVersion = ( String ) xPath.evaluate("/project/version", pom.getDocumentElement(), XPathConstants.STRING);
+
+          String convertedConfig = modifyConfigPaths( pomConfig, moduleName, moduleVersion );
+
+          addRequireJsToJar(convertedConfig, jarOutputStream);
+          
+          foundRJs = 1;
+          logger.info( "found pom.xml with requirejs config" );
+
+        } catch ( Exception e ) {
+          logger.error( "error reading pom.xml - " + e.getMessage() );
+        }
+
+      } else if ( name.endsWith( BOWER_NAME ) ) {
+
+        // next try to generate requirejs.json from bower.json (Bower WebJars)
+        if(foundRJs < 3) {
+          continue;
+        }
+
+        Matcher matcher = BOWER_PATTERN.matcher( name );
+        if ( matcher.matches() == false ) {
+          logger.error( "bower.json location isn't right" );
+          continue;
+        }
+
+        moduleName = matcher.group( 1 );
+        moduleVersion = matcher.group( 2 );
+
+        byte[] bytes = IOUtils.toByteArray( jarInputStream );
+
+        String bowerConfig = new String( bytes, "UTF-8" );
+
+        try {
+
+          String convertedConfig = requirejsFromJson( bowerConfig, moduleName, moduleVersion );
+
+          addRequireJsToJar(convertedConfig, jarOutputStream);
+          
+          foundRJs = 3;
+          logger.info( "found bower.json" );
+
+        } catch ( Exception e ) {
+          logger.error( "error reading bower.json - " + e.getMessage() );
+        }
+
+      } else if ( name.endsWith( NPM_NAME ) ) {
+
+        // next try to generate requirejs.json from package.json (NPM WebJars)
+        if(foundRJs < 4) {
+          continue;
+        }
+
+        Matcher matcher = NPM_PATTERN.matcher( name );
+        if ( matcher.matches() == false ) {
+          logger.error( "package.json location isn't right" );
+          continue;
+        }
+
+        moduleName = matcher.group( 1 );
+        moduleVersion = matcher.group( 2 );
+
+        byte[] bytes = IOUtils.toByteArray( jarInputStream );
+
+        String packageConfig = new String( bytes, "UTF-8" );
+
+        try {
+
+          String convertedConfig = requirejsFromJson( packageConfig, moduleName, moduleVersion );
+
+          addRequireJsToJar(convertedConfig, jarOutputStream);
+          
+          foundRJs = 4;
+          logger.info( "found package.json" );
+
+        } catch ( Exception e ) {
+          logger.error( "error reading package.json - " + e.getMessage() );
+        }
 
-        ZipEntry newEntry = new ZipEntry( "META-INF/js/require.json" );
-        jarOutputStream.putNextEntry( newEntry );
-        jarOutputStream.write( convertedConfig.getBytes( "UTF-8" ) );
-        // Process webjars into our form
-        jarOutputStream.closeEntry();
       } else {
-        logger.info( "copying misc entry: " + name );
```

Comment:
This change is not related with the new feature itself, but it added too much verbosity to the logs.